packages/mobile/src/firebase/saga.ts

```diff
@@ -142,4 +156,5 @@ export function* firebaseSaga() {
   yield spawn(initializeFirebase)
   yield spawn(watchLanguage)
   yield spawn(subscribeToCeloGoldExchangeRateHistory)
+  yield takeLatest(AppActions.SET_APP_STATE, setupMessaging)
```

Comment:
Conceptually, I would prefer to see this happening on a specific `APP_MOUNTED` action.
AppState API seems less natural / predictable for this, given the comment you made - that it needs to be linked to the App mounted lifecycle events.
What do you think?

Reply:
That makes sense. Should be easy to create a new action APP_MOUNTED and its inverse. Then the rest of the code stays similar. I can change it to that. Also will feel safer knowing it happens after everything is mounted for sure.